### ISSUE TITLE
Batch UpdateFromQuery queries

### ISSUE THREAD

**Issue by YeskaNova** [zzzprojects/EntityFramework-Plus#627]
Hi,

I have two update queries to run ( two UpdateFromQuery(..)), they target different tables but I'd like to batch them to reduce roundtrips to the Db. Is there any simple solution to achieve that? If not, can I get the SQL query generated for an updatefromquery ( like the standard .ToQueryString() ) so that I batch them myself ?

**Comment by JonathanMagnan** [comment]
Hello @YeskaNova ,

We will look at your request. At this moment, this is impossible/hard for both your requirement 

Best Regards,

Jon

---

**Performance Libraries**
`context.BulkInsert(list, options => options.BatchSize = 1000);  `
[Entity Framework Extensions](http://entityframework-extensions.net/) • [Entity Framework Classic](http://entityframework-classic.net/) • [Bulk Operations](http://bulk-operations.net/) •  [Dapper Plus](http://dapper-plus.net/)

**Runtime Evaluation**
`Eval.Execute("x + y", new {x = 1, y = 2}); // return 3`
[C# Eval Function](http://eval-expression.net/) • [SQL Eval Function](http://eval-sql.net/)

**Comment by JonathanMagnan** [comment]
Hello @YeskaNova ,

After some discussion, we choose to not yet add a `Batching` features.

However, we added a `CommandExecuteAction` option to let you handle it if you can do it on your side.

Here is an example:

```csharp
DbCommand dbCommand = new SqlCommand();
var c = context.EntitySimples.UpdateFromQuery(x => new EntitySimple() { ColumnInt = 20 }, option =>
{
	option.InternalIsEntityFrameworkPlus = true;
	option.BatchUpdateBuilder = opts =>
	opts.CommandExecuteAction = (command, intercepteur) =>
	{
		dbCommand = command;
		return 0;
	};
	}
);
```

This is not easy to use but in this example, the command that would have been executed is assigned to the `dbCommand` variable and we return 0 (no rows affected). You can use afterward this command to do whatever you want with the `CommandText` and `Parameters`.

**Comment by YeskaNova** [comment]
Hi @JonathanMagnan,

That's awesome, it's not simple but gives more flexibility. Thanks, I can't wait to test it. We will have it in the next release ?

**Comment by JonathanMagnan** [comment]
Oops! I forget to say it,

It's already included in the latest release ;)

**Comment by YeskaNova** [comment]
@JonathanMagnan , It seems that the code opens a connection when creating the DbCommand ( even if I'm disposing the dbcommands after that, my stress tests are blocked by that). Because I have no intention to run the DbCommand, Could we have in the options a property to tell the library to not attach the dbcommand to the DbContext connection and only return a DbCommand without a connection? That will save us thousands of DbConnection.Open() that are useless. That would be great if it's possible.

**Comment by JonathanMagnan** [comment]
We will look at and discuss this request next week.

At one point, making all this kind of customization doesn't help the library and us.

Just making the code harder to maintain on our side for being useful for a few people.

**Comment by JonathanMagnan** [comment]
Hello @YeskaNova ,

We got some discussion here about this customization and we choose that's currently out of scope.

Obviously, it would have been easy to do but at one point, we must stop to keep adding customization after customization to be able to better focus on the issue instead and improve the code readability for the maintenance.

I understand that it could be useful for you for your stress test but I believe trying to keep our code more simple and avoiding some options that we can easily avoid it a better solution.

Best Regards,

Jon

**Comment by YeskaNova** [comment]
@JonathanMagnan I agree, thank you for having considered that. BR.
